include/oneapi/dpl/pstl/hetero/dpcpp/parallel_backend_sycl_merge_sort.h

```diff
@@ -40,18 +40,39 @@ struct __subgroup_bubble_sorter
     void
     sort(const _StorageAcc& __storage_acc, _Compare __comp, std::uint32_t __start, std::uint32_t __end) const
     {
-        for (std::uint32_t i = __start; i < __end; ++i)
+        using _IndexType = std::make_signed_t<decltype(__end)>;
+
+        _IndexType __n = __end - __start;
+
+        switch (__n)
```

Comment:
Could you add a small comment justifying these extra checks?

Reply:
Done